3823 match import export header

utas-raymondng opened the pull request.

nspool commented on api/src/test/java/au/org/aodn/nrmn/restapi/service/SpreadSheetServiceIT.java:
```diff
@@ -125,6 +124,18 @@ public void shortInvertsRowShouldBeInvalid() throws Exception {
+    void rejectFileWhenRow2IsNotEmpty() {
+        Exception e = assertThrows(Exception.class, () -> {
+            FileSystemResource file3 = new FileSystemResource("src/test/resources/sheets/row2NotEmpty.xlsx");
+            ParsedSheet parsedSheet = sheetService
```
Unused variable?

utas-raymondng replied:
Yes, removed as the code expected exception

nspool commented on api/src/main/java/au/org/aodn/nrmn/restapi/service/SurveyContentsHandler.java:
```diff
@@ -14,8 +15,8 @@
 
-    private final List<String> header1Required;
-    private final List<String> header1Optional;
+    private final List<Field> header1Required;
```
`Field` is a bit too general, could it be renamed `SurveyField` instead?

nspool replied:
`Field` is a bit too general, could it be renamed `SurveyField` instead?

nspool reviewed (changes requested):
It should be possible to add the blank row using the `prependContent` property. See: https://www.ag-grid.com/react-data-grid/excel-export-extra-content/

utas-raymondng commented:
> It should be possible to add the blank row using the `prependContent` property. See: https://www.ag-grid.com/react-data-grid/excel-export-extra-content/

Correct, implemented

Regarding the ; at the bottom of the page, it is due to ; appears in the ValidationJob.js. I fixed it together in this change.

nspool reviewed (changes requested):
Getting this error when trying to reingest an exported sheet: `Row 1 missing headers: 50. Row 1 has unexpected headers: UNKNOWN`.

Please also move the Field enum into a separate file and rename it something like `SurveyField`.

utas-raymondng commented on api/src/test/java/au/org/aodn/nrmn/restapi/service/SpreadSheetServiceIT.java:
```diff
@@ -125,6 +124,18 @@ public void shortInvertsRowShouldBeInvalid() throws Exception {
+    void rejectFileWhenRow2IsNotEmpty() {
+        Exception e = assertThrows(Exception.class, () -> {
+            FileSystemResource file3 = new FileSystemResource("src/test/resources/sheets/row2NotEmpty.xlsx");
+            ParsedSheet parsedSheet = sheetService
```
> Getting this error when trying to reingest an exported sheet: `Row 1 missing headers: 50. Row 1 has unexpected headers: UNKNOWN`.
> 
> Please also move the Field enum into a separate file and rename it something like `SurveyField`.

The missing headers error is due to a mismatch type in constant.js, I fixed it and make a conversion in export function so that the header will always be type 'String' 

SurveyField refactored and become a standalone file
